src/query/response/response.ts

```diff
@@ -0,0 +1,124 @@
+export interface QueryAgentResponse {
```

Comment:
IMO, it is a better UX if `type` is used rather than `interface` to improve IDE auto-completion and code discovering. In VScode, for example, hovering an interface won't show its fields